syntax = "proto3";

package mapreduce;

message MapperInput {
  string startIndex = 1;
  string endIndex = 2;
  string oldCentroids = 3;
  string numReducer = 4;
}

message MapperOutput {
  bool success = 1;
}

message ReduceRequest {
  string reducerId = 1;
  string numMapper = 2;
}

message ReduceResponse {
  string updated_centroid = 1;
}